browser-test/src/support/admin_programs.ts

```diff
@@ -149,27 +149,22 @@ export class AdminPrograms {
   }
 
   async editProgramBlockWithOptional(programName: string, blockDescription = 'screen description', questionNames:
-  string[], optionalQuestionName: string) {
+    string[], optionalQuestionName: string) {
     await this.gotoDraftProgramEditPage(programName);
 
     await this.page.click('text=Manage Questions');
     await waitForPageJsLoad(this.page);
     await this.expectProgramBlockEditPage(programName);
-    await waitForPageJsLoad(this.page);
 
-    await this.page.click('#block-description-modal-button');
-    await waitForPageJsLoad(this.page);
+    await clickAndWaitForModal(this.page, 'block-description-modal');
     await this.page.fill('textarea', blockDescription);
-    await this.page.click('#update-block-button');
-    await waitForPageJsLoad(this.page);
+    await this.page.click('#update-block-button:not([disabled])');
 
     // Add the optional question
     await this.page.click(`button:text("${optionalQuestionName}")`);
-    await waitForPageJsLoad(this.page);

```

Comment:
Might be my bad if I asked you to remove this one, but I checked and when you click to add questions to the block it triggers a submit, so there is a page navigation and we should leave the `waitForPageJsLoad` in.

Reply:
Added that back in.  Thanks for the prompt review and for being so thorough with the browser tests!